nsqd/nsqd.go

```diff
@@ -477,8 +489,24 @@ func (n *NSQD) GetTopic(topicName string) *Topic {
 		n.Unlock()
 		return t
 	}
+	if atomic.LoadInt32(&n.isDraining) == 1 {
+		// don't create new topics when nsqd is draining
+		return nil
```

Comment:
I generally agree this is the most non-obvious bit of logic - I'll start by renaming these to `GetOrCreate....` and docstring that creation may not succeed. 

I think despite being atomic we want the `isDraining` here in the read lock so this can't race a concurrent nsqd starting to drain.

Reply:
Good point on the race. Still, we shouldn't abuse `nil` here. This is a new edge case where a topic won't be returned and we have to modify all call sites anyway to check for `nil`, might as well do it "right" and add a proper `error` return value?

Reply:
PTAL - I think the rename to `GetOrCreate{Topic,Channel}` helped a lot; I also documented that these may or may not succeed. Given that, checking nil feels an idiomatic way to implement.

We could switch but feels like bike shedding? The rename was helpful in ensuring i've covered all code paths (a few more were updated) - i should have done a more exhaustive check originally.